modules/mcc/src/utils.hpp

```diff
@@ -77,10 +78,26 @@ Mat rgb2gray(const Mat& rgb);
     @param lambda a for operation
  */
 template <typename F>
-Mat elementWise(const Mat& src, F&& lambda)
+Mat elementWise(const Mat& src, F&& lambda, Mat dst=Mat())
 {
-    Mat dst = src.clone();
+    if (dst.empty() || !dst.isContinuous() || dst.total() != src.total() || dst.type() != src.type())
+        dst = Mat(src.rows, src.cols, src.type());
     const int channel = src.channels();
+    if (src.isContinuous()) {
+        const int num_elements = (int)src.total()*channel;
+        const double *psrc = (double*)src.data;
+        double *pdst = (double*)dst.data;
+        const int batch = 128;
```

Comment:
This "batch" optimization improve performance in Windows

Reply:
Which are common values of `num_elements`? We can make `batch` dependent on number of threads:
```cpp
const int batch = num_elements / max(1, getNumThreads());
```
or
```cpp
const int batch = num_elements / (getNumThreads() > 1 ? getNumThreads() * 4 : 1);
```
instead of 4 you may choose another constant to get128 in you configuration.

Reply:
Now:
`const int batch = std::max(1, getNumThreads() > 1 ? num_elements / (1024*getNumThreads()) : num_elements);`
Now each thread processes a minimum of 1024 array elements.

With case `const int batch = num_elements / max(1, getNumThreads());` a regression in performance appears (from 47 ns ti 57 ms).

Reply:
I would suggest using batch 128, but your second sample would also work.

Reply:
Batch - the minimum required number of consecutive elements in an array that a thread can process at one time.